src/azul/service/avro_pfb.py

```diff
@@ -384,9 +389,27 @@ def _avro_pfb_schema(azul_avro_schema: Iterable[JSON]) -> JSON:
     null_int: ['string', 'long'],
     null_str: ['string'],
     null_int_sum_sort: ['string', 'long']
+    # null_datetime is handled with a custom logical type 'avro_datetime'
 }
 
 
+def datetime_to_string(data, *args):
+    return format_dcp2_datetime(data)
+
+
+def string_to_datetime(data, *args):
+    return parse_dcp2_datetime(data)
+
+
+# Keys have "<avro_type>-<logical_type>" syntax
+fastavro.write.LOGICAL_WRITERS["string-avro_datetime"] = datetime_to_string
+
+
+# Instead of converting back to a `datetime` we just keep the value in its
+# `string` form since `datetime` objects aren't JSON serializable.
```

Comment:
```suggestion
# string form since `datetime` objects aren't JSON serializable.
```

or 

```suggestion
# `str` form since `datetime` objects aren't JSON serializable.
```